features/storage/csi_clone.feature

```diff
@@ -41,4 +41,96 @@ Feature: CSI clone testing related feature
+    Then the step should succeed
+    Given I obtain test data file "storage/misc/pod.yaml"
+    When I run oc create over "pod.yaml" replacing paths:
+      | ["metadata"]["name"]                                         | mypod-clone |
+      | ["spec"]["volumes"][0]["persistentVolumeClaim"]["claimName"] | mypvc-clone |
+      | ["spec"]["containers"][0]["volumeMounts"][0]["mountPath"]    | /mnt/local  |
+    Then the step should succeed
+    Given the pod named "mypod-clone" becomes ready
+    And the "mypvc-clone" PVC becomes :bound 
+    Given the expression should be true> pv(pvc("mypvc-clone").volume_name).capacity_raw(cached: false) == "3Gi"
+    When I execute on the "mypod-clone" pod:
+      | sh | -c | ls /mnt/local/testfile |
+    Then the step should succeed
+    When I execute on the "mypod-clone" pod:
+      | sh | -c | more /mnt/local/testfile |
+    Then the step should succeed
+    And the output should contain "clone test"
+    # Need update when the filesystem is also 3Gi size(BZ) 
+
+  # @author wduan@redhat.com
+  # @case_id OCP-27690
+  Scenario: [Cinder CSI Clone] Clone a pvc with capacity less than original pvc will fail
+    Given I have a project
+    # Create mypvc-ori with 3Gi size
+    Given I obtain test data file "storage/misc/pvc.json"
+    When I create a dynamic pvc from "pvc.json" replacing paths:
+      | ["metadata"]["name"]                         | mypvc-ori    |
+      | ["spec"]["storageClassName"]                 | standard-csi |
+      | ["spec"]["resources"]["requests"]["storage"] | 3Gi          |
+    Then the step should succeed
+    Given I obtain test data file "storage/misc/pod.yaml"
+    When I run oc create over "pod.yaml" replacing paths:
+      | ["metadata"]["name"]                                         | mypod-ori  |
+      | ["spec"]["volumes"][0]["persistentVolumeClaim"]["claimName"] | mypvc-ori  |
+      | ["spec"]["containers"][0]["volumeMounts"][0]["mountPath"]    | /mnt/local |
+    Then the step should succeed
+
+    # Clone mypvc-ori with 1Gi size failed
+    Given I obtain test data file "storage/csi/pvc-clone.yaml"
+    When I create a dynamic pvc from "pvc-clone.yaml" replacing paths:
+      | ["metadata"]["name"]                         | mypvc-clone  |
+      | ["spec"]["storageClassName"]                 | standard-csi |
+      | ["spec"]["resources"]["requests"]["storage"] | 1Gi          |
+    Then the step should succeed
+    Given I obtain test data file "storage/misc/pod.yaml"
+    When I run oc create over "pod.yaml" replacing paths:
+      | ["metadata"]["name"]                                         | mypod-clone |
+      | ["spec"]["volumes"][0]["persistentVolumeClaim"]["claimName"] | mypvc-clone |
+      | ["spec"]["containers"][0]["volumeMounts"][0]["mountPath"]    | /mnt/local  |
+    Then the step should succeed
+    Given 30 seconds have passed
+    And the "mypvc-clone" PVC status is :pending
```

Comment:
@duanwei33 is this `Pending` state will persist or just a transient state?  Catching transient state is problematic.

Reply:
Thanks, updated to use the waiting loop.